src/Extensions/Hosting.Services.Web/Middlewares/UserIdentity/IdentityProviders/IUserIdentityProvider.cs

```diff
@@ -11,6 +11,11 @@ namespace Microsoft.Omex.Extensions.Hosting.Services.Web.Middlewares
 	/// </summary>
 	public interface IUserIdentityProvider
 	{
+		/// <summary>
+		/// Extracts user identity from the http context
+		/// </summary>
+		UserIdentity GetUserIdentity(HttpContext httpContext);
```

Comment:
Could you please explain a bit why this method needed. 
Initial idea was that `TryWriteBytes` provides the same information as a string in `UserIdentity.User`, just with less allocations, and that middleware should not care where are we getting user identity so I'm not sure why would we use `UserIdentity.UserHashType`?

Reply:
It's currently added as a prefix of user hash, e.g. userIdType_hashedUser. If there's an implementation which doesn't need it, it can never call this method.

Reply:
Also having this enum means that you won't be able to register in DI other implementation of `IUserIdentityProvider` that use different identity type

Reply:
Do you think having this prefix would be useful in any service?

Reply:
I don't believe it is directly used in any services.
As far as I'm aware, it was initially used for testing the user-hash generation as it was an easy mechanism to do so.
It may also be used in some analytics streams - though this could (and should) really be addressed by other means.